src/runtime_src/hip/api/hip_error_state.cpp

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Advanced Micro Device, Inc. All rights reserved.
+
+#ifdef _WIN32
+#pragma warning(disable : 4201)
+#pragma warning(disable : 4100)
+#endif
+
+#include <string>
+
+#include "hip/config.h"
+#include "hip/hip_runtime_api.h"
+#include "hip/core/device.h"
+#include "core/common/error.h"
+#include "hip/core/error_state.h"
+
+namespace xrt::core::hip
+{
+
+  static hipError_t
+  hip_peek_last_error()
+  {
+    hipError_t last_error = xrt::core::hip::error_state::GetInstance()->peek_last_error();
```

Comment:
We are already in xrt::core::hip namespace. Please dont use it again. Fix it in all places